tt_eager/tt_dnn/op_library/moreh_nll_loss_unreduced_backward/moreh_nll_loss_unreduced_backward_op.cpp

```diff
@@ -0,0 +1,172 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "tt_dnn/op_library/moreh_nll_loss_unreduced_backward/moreh_nll_loss_unreduced_backward_op.hpp"
+
+#include "tt_dnn/op_library/moreh_helper_functions.hpp"
+#include "tt_dnn/op_library/run_operation.hpp"
+#include "tt_dnn/op_library/work_split.hpp"
+#include "tt_metal/common/constants.hpp"
+#include "tt_metal/host_api.hpp"
+
+using namespace tt::constants;
+using namespace std;
+using namespace tt::tt_metal;
+
+namespace tt {
+
+namespace {
+
+inline void check_tensor(const Tensor &tensor, const std::string &op_name) {
+    TT_FATAL(tensor.get_layout() == Layout::TILE, "{} only supports tiled layout.", op_name);
+    TT_FATAL(tensor.get_dtype() == DataType::BFLOAT16, "{} only supports bfloat16.", op_name);
+    TT_FATAL(
+        tensor.storage_type() == StorageType::DEVICE, "Operands to {} need to be on device!", op_name);
+    TT_FATAL(
+        tensor.buffer() != nullptr, "Operands to {} need to be allocated in buffers on device!", op_name);
+}
+
+inline void check_tensor(std::optional<Tensor> tensor, const std::string &op_name) {
+    if (!tensor.has_value()) {
+        return;
+    }
+    check_tensor(tensor.value(), op_name);
+}
+
+}  // namespace
+
+namespace operations {
+namespace primary {
+
+void MorehNllLossUnreducedBackward::validate_with_output_tensors(
+    const std::vector<Tensor>& input_tensors,
+    const std::vector<std::optional<const Tensor>>& optional_input_tensors,
+    const std::vector<std::optional<Tensor>>& output_tensors) const {
+    TT_FATAL(input_tensors.size() == 2, "Must have 2 input tensors");
+    TT_FATAL(optional_input_tensors.size() == 1, "Must have 1 optional input tensors");
+
+    auto& target_tensor = input_tensors.at(0);
+    auto& output_grad_tensor = input_tensors.at(1);
+
+    auto& weight_tensor = optional_input_tensors.at(0);
+
+    auto& input_grad_tensor = output_tensors.at(0);
```

Comment:
(nit)
would it be possible to hold them by `const auto&` instead?

Reply:
```C++
const auto& target_tensor = input_tensors.at(0);
const auto& output_grad_tensor = input_tensors.at(1);

const auto& weight_tensor = optional_input_tensors.at(0);

const auto& input_grad_tensor = output_tensors.at(0);
```
I have updated it to `const auto&`